File: Prime factorization and geek number - GFG/prime-factorization-and-geek-number.py
#User function Template for python3

class Solution:
    def geekNumber(self, N):
        i = 2
        while i * i <= N:
            count = 0
            while N % i == 0:
                count += 1
                N = N / i
            if count > 1:
                return 0
            i += 1
        return 1

#{ 
#  Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input())
        
        ob = Solution()
        if ob.geekNumber(N)==1 :
            print("Yes");
        else :
            print("No");
# } Driver Code Ends

File: Prime factorization and geek number - GFG/prime-factorization-and-geek-number.cpp
// { Driver Code Starts
//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
//User function Template for C++

class Solution{
public:
    int geekNumber(int N){
         int div = 2;
          while(div*div<=N)
          {
              int counter  = 0;
              while(N%div==0)
              {
                  counter++;
                  N = N/div;
              }
              div++;
              if(counter>1)
              {
                  return 0;
              }
          }
          return 1;
    }
};

// { Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        int N;
        cin>>N;
        
        Solution ob;
        if(ob.geekNumber(N))
            cout<<"Yes\n";
        else
            cout<<"No\n";
    }
    return 0;
}  // } Driver Code Ends